### DISCUSSION TITLE
How to have default date() on a TextInput

### DISCUSSION THREAD

**Discussion by newtec-foggia** [filamentphp/filament#5510] [Help]
does not display the current date, how to ?

**TextInput::make('data_del_riscontro')
                            ->type('date')
                            ->extraInputAttributes(['value' => now()])
                            ->label('data riscontro')
                            ->required(),**

Thanks,

**Comment by ArtMin96** [comment]
Why don't you use date picker? [https://filamentphp.com/docs/2.x/forms/fields#date-time-picker](https://filamentphp.com/docs/2.x/forms/fields#date-time-picker)

**Comment by ddoddsr** [comment]
Any ideas? I'd like a text entry date with today as default

**Reply by danharrin** [reply]
`default(now()->toDateString())`

**Reply by marcelo-lipienski** [reply]
Does this actually work?

I tried using `default(now()->toDateString())` in both a `TextInput` component with a `type('date')` and a `DatePicker` component, but none worked.

I tested on Firefox and Chromium running on Linux.

**Reply by danharrin** [reply]
Make sure you're on the Create page and not the Edit page of a resource.

**Reply by marcelo-lipienski** [reply]
Nice catch. Unfortunately, I need to set up a default value on the Edit page. Would this be impossible then?

**Reply by leandrocfe** [reply]
[field-hydration](https://filamentphp.com/docs/3.x/forms/advanced#field-hydration)

**Reply by marcelo-lipienski** [reply]
Thanks @danharrin @leandrocfe 
To summarize for anyone searching for this in the future:

`default()` is not supposed to set up a default value on an EditPage.
The way to achieve a default value on an EditPage is to set up a `afterStateHydrated()`

In my case, I need to fill in a default value on an EditPage for a field that might or might not already have a value, and this is how I implemented the `afterStateHydrated` to handle those two cases:

```php
DatePicker::make('date')
    ->afterStateHydrated(function (DatePicker $component, ?string $state) {
        // if the value is empty in the database, set a default value, if not, just continue with the default component hydration
        if (!$state) {
            $component->state(now()->toDateString());
        }
    })
```

**Reply by ccc-developer** [reply]
@marcelo-lipienski, I appreciate you taking the time to detail your solution here. I had combed through the documentation and, like you mentioned, the default function wasn't clearly documented. It's moments like these that remind us of the old saying: "It's often the most obvious solutions that elude us." Thanks for sharing your solution.

**Comment by lordjack** [comment]
> Thanks @danharrin @leandrocfe To summarize for anyone searching for this in the future:
> 
> `default()` is not supposed to set up a default value on an EditPage. The way to achieve a default value on an EditPage is to set up a `afterStateHydrated()`
> 
> In my case, I need to fill in a default value on an EditPage for a field that might or might not already have a value, and this is how I implemented the `afterStateHydrated` to handle those two cases:
> 
> ```
> DatePicker::make('date')
>     ->afterStateHydrated(function (DatePicker $component, ?string $state) {
>         // if the value is empty in the database, set a default value, if not, just continue with the default component hydration
>         if (!$state) {
>             $component->state(now()->toDateString());
>         }
>     })
> ```
Thank you for your tip

**Comment by cirrion-software** [comment]
Forms\Components\DatePicker::make('date')
                    ->default(date(now()))

**Reply by cirrion-software** [reply]
Whoops sorry I didn't read the OP properly
